### PR TITLE
💖 feat(documentation): Logs, Console Logs, Environment Variables

### PR DESCRIPTION
**Pull request by lambro**
Added information on Logs and Console Logs for KintoApp Environments

### CODE REVIEW

File: docs/kintoapps.md [hunk 1]
```diff
@@ -93,11 +93,42 @@ Be sure to include in the authorization header. Example:
...
+- Which endpoint was called
+- Which KintoBlock was called
+- Which version of the KintoBlock was called
+- The time and date of the call
+
+If you wish to view more details of the call, these can be found by expanding that particular calls row in the table. Here you can view the JSON of any logs you have made. 
+
+### Required Format for Console Logs
+
+In order to view Console Logs in our system, specific syntax must be used. Every call to the block will container a header called `kinto-request-id`. This header will contain a value that needs to be sent back to us in order to associate the console log with the correct call. It is important to send your information formatted in JSON, with the correct quotation marks like this : `"key"`. An example for a *Node.js / Express* app would look like this:
```
Comment (nadeemkhedr) [inline docs/kintoapps.md]: typo `will container` i think u meant `will contain`

Comment (nadeemkhedr) [reply docs/kintoapps.md]: > This header will contain a value that needs to be sent back to us

I think that is confusing, its not gonna be send back in the response or anything, i would say it just needs to be as part of the log instead of sent back

File: docs/kintoblocks.md [hunk 2]
```diff
@@ -43,6 +43,49 @@ The steps to tag a commit are:
...
+
+Environment Variables are used to add values required for your KintoBlock to work out in the world. These are often configurations like  `CLIENT_IDS` and `SECRETS` that you do not want to expose in your code. Inside your KintoBlock you acess these via `env` variables. 
+
+In a *Node.js* KintoBlock for example these are accessed using this syntax:
+
+```
+process.env.WHATEVER_KEY_THE_DEVELOPER_USED.
+```
+
+These are accessed during the build process so when updating them you need to create a new TAG.
```
Comment (nadeemkhedr) [inline docs/kintoblocks.md]: that's not true actually, creating a tag won't trigger the build, now the only way to trigger it is by adding a commit

File: docs/kintoblocks.md [hunk 3]
```diff
@@ -43,6 +43,49 @@ The steps to tag a commit are:
...
+### Using Environment Variables
+
+The section on Environment Variables lives on the *KintoBlock Edit* page, towards the end of the form. They are specific to a *branch* of a KintoBlock, so if you switch branches and want to use that particular branch in a KintoApp you will need to input the *Environment Variables* on that branch too.
+
+### Points to remember about Environment Variables
+
+- These can only be edited by the creator of a KintoBlock.
+- If you are building a block that is shared, will be included in multiple applications, and will require information from the end user, it is best to use *Custom Paramaters*.
+
 ## Custom Parameters (Coming Soon)
```
Comment (nadeemkhedr) [inline docs/kintoblocks.md]: I would move the custom param section to another PR that is blocked and add there the database as well

Comment (nadeemkhedr) [reply docs/kintoblocks.md]: so we would be able to merge the PR

File: docs/kintoblocks.md [hunk 4]
```diff
@@ -43,6 +43,49 @@ The steps to tag a commit are:
...
-## Environment Parameters (Coming Soon)
+
+Custom Parameters are another way to pass information to your KintoApp. These values are editable by the end user in the case of shared KintoBlocks, and also editable at a KintoApp level for cases where multiple apps are using the same KintoBlock.
+
+When creating KintoBlock you can assign default values to your Custom Parameters and these can be marked required if they are necessary for your KintoBlock to function.
+
+Header values are accessible inside your code within each call, and you specify when to use them via API doc headers, in a *Node.js* application adding the key to a call will look like this:
```
Comment (nadeemkhedr) [inline docs/kintoblocks.md]: > Header values are accessible inside your code within each call

you wanna say the custom params values are accessible as headers

File: docs/kintoblocks.md [hunk 5]
```diff
@@ -43,6 +43,49 @@ The steps to tag a commit are:
...
+```
+
+This will be injected as a header into this call, and accessible in the usual way you would access headers. In a *Node.js / Express* KintoBlock it would look like this: 
+
+```
+  const githubToken = req.get('githubToken')
+```
+
+### Points to remember about Custom Parameters 
+- Use these to set variables that the customer will need to set or change, like API keys for external APIS.
```
Comment (nadeemkhedr) [inline docs/kintoblocks.md]: i would say end user instead of customer and maybe say in quotes or bracket the person who is gonna add the microservice to his app

File: docs/kintoblocks.md (hunk 3 above)
Comment (lambro) [reply docs/kintoblocks.md]: No.

File: docs/kintoapps.md [hunk 6]
```diff
@@ -93,11 +93,42 @@ Be sure to include in the authorization header. Example:
...
+- Which endpoint was called
+- Which KintoBlock was called
+- Which version of the KintoBlock was called
+- The time and date of the call
+
+If you wish to view more details of the call, these can be found by expanding that particular calls row in the table. Here you can view the JSON of any logs you have made. 
+
+### Required Format for Console Logs
+
+In order to view Console Logs in our system, specific syntax must be used. Every call to the block will contain a header called `kinto-request-id`. This header will contain a value that needs to be included in the **Console Log** in order to associate each **Console Log** with the correct call. It is important to send your information formatted in JSON, with the correct quotation marks like this : `"key": "value"`. An example for a **Node.js / Express** app would look like this:
```
Comment (Disturbing) [inline docs/kintoapps.md]: "In our system to view console logs in our system" => your system?  Its also not clear when u say every call to the blcok will contain a haeder.

--
Console logs can help you easily debug what has occurred during the lifetime of an API call.  We require a specific json format because console logs will be able to be used for alerts and analytics in the near future.

//TODO image

All console logs that do not follow our format will not be available for you to view. The format is as follows:

{ "kinto_request_id": "acd12ff21-f2f21f1wf1-2f21f1s", "customdata": "hello", "message": "world" }

Every incoming API call that comes to your service will have a unique kinto_request_id. You must log that Id into the console log as the value for "kinto_request_id".  This allows us to track your specific call and perform magic!

Here's and example of what it would look like in nodejs

//here


You may use middleware to automatically do this for each call as well. We'll provide examples of this in the near future.

File: docs/kintoapps.md [hunk 7]
```diff
@@ -93,11 +93,42 @@ Be sure to include in the authorization header. Example:
 
 To make a call to gRPC, talk to grpc.prod.kintohub.com with your gRPC service proto file, but include in the grpc-meta-data the authorization header as well.
 
+## Viewing KintoApp Logs
```
Comment (Disturbing) [inline docs/kintoapps.md]: Image of logs

File: docs/kintoblocks.md [hunk 8]
```diff
@@ -43,6 +43,28 @@ The steps to tag a commit are:
 
 The main reason to tag a KintoBlock is to add it to an application, think of it as a release, so when you have your KintoBlock finished and want to update it in an application you can tag it and select the tag in the application. This is a way to control the released versions of your KintoBlock.
 
-## Custom Parameters (Coming Soon)
-## Environment Parameters (Coming Soon)
+## Environment Variables
+
+Environment Variables are used to add values required for your KintoBlock to work out in the world. These are often configurations like  `CLIENT_IDS` and `SECRETS` that you do not want to expose in your code. Inside your KintoBlock you acess these via `env` variables.
```
Comment (Disturbing) [inline docs/kintoblocks.md]: To work out in the world => need to be more technical.

- Environment variables are used to control static information that you want to control in the application.
- Environment variables are static between multiple environments.  You cannot change an environment variables between applications and their environments.
- In the near future configuration management will be used to configure your application as a per-environment use case.

I think we need to discuss this so its more clear - cuz client ids / secrets should be config params, not environment variables.

File: docs/kintoblocks.md [hunk 9]
```diff
@@ -43,6 +43,28 @@ The steps to tag a commit are:
...
+
+Environment Variables are used to add values required for your KintoBlock to work out in the world. These are often configurations like  `CLIENT_IDS` and `SECRETS` that you do not want to expose in your code. Inside your KintoBlock you acess these via `env` variables. 
+
+In a **Node.js** KintoBlock for example these are accessed using this syntax:
+
+```
+process.env.WHATEVER_KEY_THE_DEVELOPER_USED.
+```
+
+These are accessed during the build process so when updating them you need to add a new commit to that branch (This is due to be updated soon, but currently the only way to update **Environment Variables** is to trigger a new build with a new commit).
```
Comment (Disturbing) [inline docs/kintoblocks.md]: @sanguinius69 @bappr  --- we should probably do env vars at loading, not on build because this is annoying! :)

File: docs/kintoblocks.md [hunk 10]
```diff
@@ -43,6 +43,28 @@ The steps to tag a commit are:
...
+
+### Using Environment Variables
```
Comment (Disturbing) [inline docs/kintoblocks.md]: managing, not using.

File: docs/kintoblocks.md (hunk 8 above)
Comment (Disturbing) [reply docs/kintoblocks.md]: Example would be like private keys.

File: docs/kintoblocks.md [hunk 11]
```diff
@@ -43,6 +43,28 @@ The steps to tag a commit are:
...
+
+The section on Environment Variables lives on the **KintoBlock Edit** page, towards the end of the form. They are specific to a **branch** of a KintoBlock, so if you switch branches and want to use that particular branch in a KintoApp you will need to input the **Environment Variables** on that branch too.
+
+
+### Points to remember about Environment Variables
+
+- These can only be edited by the creator of a KintoBlock.
+- If you are building a block that is shared, will be included in multiple applications, and will require information from the end user, it is best to use *Custom Paramaters*.
```
Comment (Disturbing) [inline docs/kintoblocks.md]: Nice

File: docs/kintoblocks.md [hunk 12]
```diff
@@ -43,6 +43,28 @@ The steps to tag a commit are:
...
+
+
+## Custom Parameters (coming soon)
```
Comment (Disturbing) [inline docs/kintoblocks.md]: Maybe we should describe custom params, because ENV vars are not making sense as well without knowing both of them, (even though its coming soon_)

Comment (lambro) [reply docs/kintoblocks.md]: The PR for custom params is also ready, shall we merge it with "coming soon" and link to it here?
